File: pinyougou-interface/src/main/java/cn/itcast/core/service/BrandService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.good.Brand;
import entity.PageResult;
import org.apache.poi.hssf.usermodel.HSSFWorkbook;
import org.springframework.web.bind.annotation.ModelAttribute;

import javax.servlet.http.HttpServletResponse;
import java.util.List;
import java.util.Map;

public interface BrandService {

    //查询所有品牌
    public List<Brand> findAll();
    //查询分页对象
    PageResult findPage(Integer pageNum, Integer pageSize);

    //保存
    void add(Brand brand);

    Brand findOne(Long id);

    void update(Brand brand);

    void deletes(Long[] ids);

    PageResult search(Integer pageNum, Integer pageSize, Brand brand);

    List<Map> selectOptionList();
    List<Brand> seleExecle();

    List<Map> showList(List<Map<String, String>> tt);

    void insertExcel(String s);
}

File: pinyougou-web-manager/src/main/webapp/js/service/brandService.js
// 定义服务层:
app.service("brandService",function($http){
	this.findAll = function(){
		return $http.get("../brand/findAll.do");
	}
	
	this.findPage = function(page,rows){
		return $http.get("../brand/findPage.do?pageNum="+page+"&pageSize="+rows);
	}
	
	this.add = function(entity){
		return $http.post("../brand/add.do",entity);
	}
	
	this.update=function(entity){
		return $http.post("../brand/update.do",entity);
	}
	
	this.findOne=function(id){
		return $http.get("../brand/findOne.do?id="+id);
	}
	
	this.dele = function(ids){
		return $http.get("../brand/deletes.do?ids="+ids);
	}
	
	this.search = function(page,rows,searchEntity){
		return $http.post("../brand/search.do?pageNum="+page+"&pageSize="+rows,searchEntity);
	}
	
	this.selectOptionList = function(){
		return $http.get("../brand/selectOptionList.do");//url 入参：无  返回值：List<Map>
	}

    this.updateStatus = function(id,ids){
        return $http.get("../brand/updateStatus.do?id="+id+"&ids="+ids);//id: 状态
    }
});